vllm/engine/llm_engine.py

```diff
@@ -255,6 +255,9 @@ def add_request(
         if prompt_token_ids is None:
             assert prompt is not None
             prompt_token_ids = self.tokenizer.encode(prompt)
+        if prompt is None:
+            prompt = self.tokenizer.decode(prompt_token_ids,
+                                           skip_special_tokens=True)
```

Comment:
Why is this necessary?

Reply:
If the user only passes `prompt_token_ids`, the output won't be able to echo the prompt in its string form since it is not provided; it isn't the case vice versa.

Reply:
If the user only passes prompt_token_ids, the output won't be able to echo the prompt in its string form since it is not provided; it isn't the case vice versa.